lib/system/syslocks.nim

```diff
@@ -102,26 +102,18 @@ else:
     SysLockObj {.importc: "pthread_mutex_t", pure, final,
                header: """#include <sys/types.h>
                           #include <pthread.h>""".} = object
-      when defined(linux) and defined(amd64):
-        abi: array[40 div sizeof(clong), clong]
 
     SysLockAttr {.importc: "pthread_mutexattr_t", pure, final
                header: """#include <sys/types.h>
                           #include <pthread.h>""".} = object
-      when defined(linux) and defined(amd64):
-        abi: array[4 div sizeof(cint), cint]  # actually a cint
 
     SysCondObj {.importc: "pthread_cond_t", pure, final,
                header: """#include <sys/types.h>
                           #include <pthread.h>""".} = object
-      when defined(linux) and defined(amd64):
-        abi: array[48 div sizeof(clonglong), clonglong]
 
     SysCondAttr {.importc: "pthread_condattr_t", pure, final
                header: """#include <sys/types.h>
                           #include <pthread.h>""".} = object
-      when defined(linux) and defined(amd64):
-        abi: array[4 div sizeof(cint), cint]  # actually a cint
```

Comment:
Changes like these break `nlvm` (which is still our best idea of how to get good debugging support btw).

Reply:
but why single out this type? other types have no guarantee to have all their fields listed in the importc type (unless `{.completeStruct.}` is given).

furthermore, nlvm is heavily behind nim devel: last nim commit that was integrated is Jul 29, 2020, https://github.com/arnetheduck/Nim/commit/bf320ed172f74f60fd274338e82bdc9ce3520dd9)

and there are other ways to do this without resorting to platform specific conditional fields. 

abi/padding fields that aren't used just shouldn't be exposed, which leads to bugs like #14873 etc

Reply:
yes, that would be better than `abi` field, but it still causes the same problems because the ABI for those is not specified and varies by OS/distribution.

note that you can still use `sizeof` at RT; `{.sizeof: N.}` only makes sense if you need it at CT (unlikely because, well, VM is single threaded).

Note also that on osx i get:
```nim
import std/locks
var a: Lock
echo a.sizeof # 64
```
even though no nim abi field is there in the definition of `SysLockObj`, which shows how arbitrary this was. 

Long story short: removing the `abi`, as i did in this PR, makes sense.

Reply:
@araq also you can always do this if you need to access raw bytes:

```
import std/[sugar,locks]

proc getBytes[T](buf: var seq[uint8], a: T) =
  let n = T.sizeof
  buf.setLen n
  echo n
  copyMem(buf[0].addr, a.unsafeAddr, n)

var a: Lock
assert a.sizeof == 64
initLock(a)

echo seq[uint8].default.dup(getBytes(a)) # @[90, 84, 85, 77, 0, 0, 0, 0, ... 85, 77, 90, 84, 85, 77]
```

Reply:
The abi field never was about accessing bytes, it was about getting sizeof and alignof right.

Reply:
> need it at CT

you need it for pointer arithmetic, memory allocation, offset arithmetic and a number of other cases - this PR breaks existing code and introduces differences between compile time and runtime that lead to bugs.

The proper fix is to provide the Nim compiler with correct information at compile time - this can indeed be through libclang - or by just spelling it out.